geo/src/algorithm/line_interpolate_point.rs

```diff
@@ -84,7 +84,7 @@ where
                 let line_frac = (fractional_length - x.0) / x.2;
                 (x.3).line_interpolate_point(&line_frac)
             }
-            None => self.points_iter().last().unwrap(),
+            None => self.points_iter().last().unwrap_or(Point::new(T::nan(), T::nan())),
```

Comment:
As your test shows, and comment dictates, does this happen if and only if line_string is empty? No other cases?

panic'ing isn't great, but Rust offers some other more expressive tools than sentinel values. I wonder if returning an Option or Result would make more sense.

Would making output an Option or a Result type be better?

Reply:
I agree - perhaps an enum in the vein of (geo::Closest)[https://docs.rs/geo/0.15.0/geo/enum.Closest.html]?